usaspending_api/common/query_with_filters.py

```diff
@@ -624,26 +622,44 @@ class _ProgramActivities(_Filter):
     underscore_name = "program_activities"
 
     @classmethod
-    def generate_elasticsearch_query(cls, filter_values: List[str], query_type: _QueryType, **options) -> ES_Q:
-        award_id_match_query = []
+    def generate_elasticsearch_query(cls, filter_values: List[dict], query_type: _QueryType, **options) -> ES_Q:
+        program_activity_match_queries = []
+
         for filter_value in filter_values:
-            query_filter_predicates = [Q(program_activity_id__isnull=False)]
-
-            if "name" in filter_value:
-                query_filter_predicates.append(Q(program_activity__program_activity_name=filter_value["name"].upper()))
-            if "code" in filter_value:
-                query_filter_predicates.append(Q(program_activity__program_activity_code=filter_value["code"]))
-            award_ids_filtered_by_program_activities = FinancialAccountsByAwards.objects.filter(
-                *query_filter_predicates
-            )
-            award_ids = set(award_ids_filtered_by_program_activities.values_list("award_id", flat=True))
+            if "name" in filter_value and "code" in filter_value:
+                program_activity_match_queries.append(
+                    ES_Q(
```

Comment:
Not worth blocking over, but this could be cleaned up since you are repeating yourself a bit. For example, build a list of the `must` values and then at the end append the `ES_Q()` object. You could then simplify the conditionals down to check the `name`, check the `code`, and if the `must` list has values then append the `ES_Q`.